### ISSUE TITLE
ComputeTargetHTTPProxy appears to assume urlMapRef is global

### ISSUE THREAD

**Issue by nstogner** [GoogleCloudPlatform/k8s-config-connector#594]
## Bug Description

I am trying to implement an Internal Load Balancer HTTP to HTTPS redirect using Config Connector.

I am running into issues where KCC appears to look for a global UrlMap when I specified a regional one (via `location: us-central1`):

```
Error 404: The resource 'projects/my-project/global/urlMaps/https-redirect-foo' was not found, notFound
```

## Additional Diagnostic Information

Output of `kubectl describe`:

```
Name:         https-redirect-proxy-foo
Namespace:    cc
Labels:       <none>
Annotations:  cnrm.cloud.google.com/management-conflict-prevention-policy: none
              cnrm.cloud.google.com/project-id: my-project
              cnrm.cloud.google.com/state-into-spec: merge
API Version:  compute.cnrm.cloud.google.com/v1beta1
Kind:         ComputeTargetHTTPProxy
Metadata:
  Creation Timestamp:  2022-01-19T16:46:14Z
  Finalizers:
    cnrm.cloud.google.com/finalizer
    cnrm.cloud.google.com/deletion-defender
  Generation:  1
  Managed Fields:
    API Version:  compute.cnrm.cloud.google.com/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:location:
        f:proxyBind:
        f:urlMapRef:
          .:
          f:external:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2022-01-19T16:46:14Z
    API Version:  compute.cnrm.cloud.google.com/v1beta1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"cnrm.cloud.google.com/deletion-defender":
          v:"cnrm.cloud.google.com/finalizer":
      f:status:
        .:
        f:conditions:
        f:observedGeneration:
    Manager:         cnrm-controller-manager
    Operation:       Update
    Time:            2022-01-19T16:46:15Z
  Resource Version:  2703582
  UID:               5ebf51b8-8e77-4183-a043-afedec6c48b2
Spec:
  Location:    us-central1
  Proxy Bind:  true
  URL Map Ref:
    External:  https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/urlMaps/https-redirect-foo
Status:
  Conditions:
    Last Transition Time:  2022-01-19T16:46:15Z
    Message:               Update call failed: error applying desired state: summary: Error creating TargetHttpProxy: googleapi: Error 404: The resource 'projects/my-project/global/urlMaps/https-redirect-foo' was not found, notFound
    Reason:                UpdateFailed
    Status:                False
    Type:                  Ready
  Observed Generation:     1
```

### Kubernetes Cluster Version

Client Version: v1.22.4
Server Version: v1.21.5-gke.1302

### Config Connector Version

1.63.0

### Config Connector Mode

cluster

## Log Output

```json
{"severity":"error","timestamp":"2022-01-19T20:51:25.778Z","logger":"controller.computetargethttpproxy-controller","msg":"Reconciler error","reconciler group":"compute.cnrm.cloud.google.com","reconciler kind":"ComputeTargetHTTPProxy","name":"https-redirect-proxy-foo","namespace":"cc","error":"Update call failed: error applying desired state: summary: Error creating TargetHttpProxy: googleapi: Error 404: The resource 'projects/my-project/global/urlMaps/https-redirect-foo' was not found, notFound"}
```

### Steps to reproduce the issue
Apply the yaml below.

### YAML snippets

```yaml
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeBackendService
metadata:
  name: https-redirect-empty-backend-foo
spec:
  loadBalancingScheme: INTERNAL_MANAGED
  location: us-central1
  protocol: HTTP
  logConfig:
    enable: false
---
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeURLMap
metadata:
  name: https-redirect-foo
spec:
  location: us-central1
  defaultService:
    backendServiceRef:
      name: https-redirect-empty-backend-foo
  hostRule:
    - hosts:
        - "*"
      pathMatcher: "all"
  pathMatcher:
    - name: "all"
      defaultUrlRedirect:
        hostRedirect: 10.128.0.33:443
        pathRedirect: /
        redirectResponseCode: PERMANENT_REDIRECT
        httpsRedirect: true
        stripQuery: false
---
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeTargetHTTPProxy
metadata:
  name: https-redirect-proxy-foo
spec:
  proxyBind: true
  location: us-central1
  urlMapRef:
    name: https-redirect-foo
    # NOTE: Also attempted to more directly reference the url map...
    #namespace: cc
    #external: https://www.googleapis.com/compute/v1/projects/my-project/regions/us-central1/urlMaps/https-redirect-foo
    #external: projects/my-project/regions/us-central1/urlMaps/https-redirect-foo
---
apiVersion: compute.cnrm.cloud.google.com/v1beta1
kind: ComputeForwardingRule
metadata:
  name: https-redirect-foo
spec:
  target:
    targetHTTPProxyRef:
      name: https-redirect-proxy-foo
  location: us-central1
  ipAddress:
    addressRef:
      name: nick-static-ilb # TODO: Create with ConfigConnector
```

**Comment by nstogner** [comment]
I forgot to mention, the same behavior occurs when using `spec.urlMapRef.external` in ComputeTargetHTTPProxy  with a link that contains `.../regions/us-central1/...`.

**Comment by jcanseco** [comment]
Hey @nstogner, thank you for filing this issue! It seems this has actually been fixed in KCC v1.71.0 which we've [just released](https://github.com/GoogleCloudPlatform/k8s-config-connector/releases/tag/v1.71.0).

Closing this issue out, but please feel free to re-open if needed.
